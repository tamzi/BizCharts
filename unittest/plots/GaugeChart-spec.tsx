import React from 'react';
import GaugeChart from '../../src/plots/GaugeChart';
import { render, cleanup } from '@testing-library/react';

const MOCK_DATA = [
  { action: '浏览网站', pv: 50000, quarter: '2020Q1' },
  { action: '放入购物车', pv: 35000, quarter: '2020Q1' },
  { action: '生成订单', pv: 25000, quarter: '2020Q1' },
  { action: '支付订单', pv: 15000, quarter: '2020Q1' },
  { action: '完成交易', pv: 11500, quarter: '2020Q1' },
  { action: '浏览网站', pv: 80000, quarter: '2020Q2' },
  { action: '放入购物车', pv: 63000, quarter: '2020Q2' },
  { action: '生成订单', pv: 47000, quarter: '2020Q2' },
  { action: '支付订单', pv: 24000, quarter: '2020Q2' },
  { action: '完成交易', pv: 17500, quarter: '2020Q2' },
];


describe('Plots-GaugeChart', () => {
  test('旧版仪表图', () => {
    let chart = null;
    render(<GaugeChart
      title={{
        visible: true,
        text: '旧版仪表图',
      }}
      width={400}
      height={400}
      value={78}
      min={0}
      max={100}
      // @ts-ignore
      range={[0, 1 / 3, 2 / 3, 1]}
      color={['#F4664A', '#FAAD14', '#30BF78']}
      statistic={{
        text: '优',
        color: '#30bf78',
      }}
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
  test('基础仪表图', () => {
    let chart = null;
    render(<GaugeChart
      title={{
        visible: true,
        text: '基础仪表图',
      }}
      width={400}
      height={400}
      percent={0.75}
      // @ts-ignore
      range={{
        ticks: [0, 1 / 3, 2 / 3, 1],
        color: ['#F4664A', '#FAAD14', '#30BF78'],
      }}
      onGetG2Instance={
        (c) => chart = c
      }
    />)
    expect(chart.options).toMatchSnapshot();
    cleanup();
  })
})
