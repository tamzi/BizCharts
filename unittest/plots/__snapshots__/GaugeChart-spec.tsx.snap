// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plots-GaugeChart 基础仪表图 1`] = `
Object {
  "animation": false,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "axis": Object {
    "grid": null,
    "label": Object {
      "offset": -24,
      "style": Object {
        "textAlign": "center",
        "textBaseline": "middle",
      },
    },
    "line": null,
    "subTickLine": Object {
      "length": -8,
    },
    "tickLine": Object {
      "length": -12,
    },
  },
  "data": 0.75,
  "endAngle": 0.5235987755982988,
  "height": 400,
  "indicator": Object {
    "pin": Object {
      "style": Object {
        "fill": "#fff",
        "lineWidth": 4.5,
        "r": 9.75,
        "stroke": "#D0D0D0",
      },
    },
    "pointer": Object {
      "style": Object {
        "lineCap": "round",
        "lineWidth": 5,
        "stroke": "#D0D0D0",
      },
    },
  },
  "innerRadius": 0.9,
  "meta": Object {
    "percent": Object {
      "sync": "v",
      "tickCount": 5,
      "tickInterval": 0.2,
    },
    "range": Object {
      "sync": "v",
    },
  },
  "percent": 0.75,
  "radius": 0.95,
  "range": Object {
    "color": Array [
      "#F4664A",
      "#FAAD14",
      "#30BF78",
    ],
    "ticks": Array [
      0,
      0.3333333333333333,
      0.6666666666666666,
      1,
    ],
  },
  "startAngle": -3.6651914291880923,
  "statistic": Object {
    "title": false,
  },
  "syncViewPadding": true,
  "width": 400,
}
`;

exports[`Plots-GaugeChart 旧版仪表图 1`] = `
Object {
  "animation": false,
  "appendPadding": Array [
    10,
    5,
    10,
    10,
  ],
  "autoFit": true,
  "axis": Object {
    "grid": null,
    "label": Object {
      "formatter": [Function],
      "offset": -24,
      "style": Object {
        "textAlign": "center",
        "textBaseline": "middle",
      },
    },
    "line": null,
    "subTickLine": Object {
      "length": -8,
    },
    "tickLine": Object {
      "length": -12,
    },
  },
  "color": Array [
    "#F4664A",
    "#FAAD14",
    "#30BF78",
  ],
  "data": 0.78,
  "endAngle": 0.5235987755982988,
  "height": 400,
  "indicator": Object {
    "pin": Object {
      "style": Object {
        "fill": "#fff",
        "lineWidth": 4.5,
        "r": 9.75,
        "stroke": "#D0D0D0",
      },
    },
    "pointer": Object {
      "style": Object {
        "lineCap": "round",
        "lineWidth": 5,
        "stroke": "#D0D0D0",
      },
    },
  },
  "innerRadius": 0.9,
  "meta": Object {
    "percent": Object {
      "sync": "v",
      "tickCount": 5,
      "tickInterval": 0.2,
    },
    "range": Object {
      "sync": "v",
    },
  },
  "percent": 0.78,
  "radius": 0.95,
  "range": Object {
    "color": Array [
      "#F4664A",
      "#FAAD14",
      "#30BF78",
    ],
    "ticks": Array [
      0,
      0.003333333333333333,
      0.006666666666666666,
      0.01,
    ],
  },
  "startAngle": -3.6651914291880923,
  "statistic": Object {
    "color": "#30bf78",
    "text": "优",
    "title": false,
  },
  "syncViewPadding": true,
  "width": 400,
}
`;
